src/config.py

```diff
@@ -13,11 +13,19 @@ def __init__(self):
     self._json = {}
 
   def load(self, config_filepath: str):
```

Comment:
Thank you for this! I've had a thought that's a _bit_ more of a refactor but I think it'd be a net-positive change overall.

How about instead of taking in the path to a file, the class' initializer simply takes in a dictionary (or many dictionaries?) representing the configuration data. This way, you can feed in _anything_ as a config source - a file, env vars, CLI args (in the future) and this module just exposes an API to interact with the config.

High-level pseudocode:

```python
class Config:
  def __init__(self, *args)
    self._config = {}
    # Args is a list of dicts
    for arg in args:
      self._config.merge(downcase_keys(args))
```

And then in`main.py` there could be a method like

```python
def __build_configuration():
  file_config = contents if file provided else {}
  env_vars = ...
  
  return Config(file_config, env_vars)
```